tests/text/test_eed.py

```diff
@@ -0,0 +1,79 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from torchmetrics.functional.text.eed import eed
+from torchmetrics.text.eed import EED
+
+HYPOTHESES_1 = "perfect match"
+REFERENCES_1 = "perfect match"
+
+HYPOTHESES_2 = "imperfect match"
+REFERENCES_2 = "blue orange"
+
+# test batches
+BATCH_1 = {"hypotheses": [HYPOTHESES_1, HYPOTHESES_2], "references": [REFERENCES_1, REFERENCES_2]}
+
```

Comment:
I've already started differing from the original implementation. They originally used:

```python
row = [1] * (len(hyp) + 1)
row[0] = 0  # CDER initialisation 0,0 = 0, rest 1
```

...which leads to different results when the order of sentences are different. You don't see this until you run things in parallel vs the original implementation. I've simulated this scenario with `test_parallelisation_eed`. The only reason it's using `np.isclose` vs `assert _ equals _` is because ubuntu does not pass this test while windows and mac does.

Currently, I'm using:

```
row = [1.0] * (len(hyp) + 1)
row[0] = 0.0  # CDER initialisation 0,0 = 0.0, rest 1.0
```

... which has consistent results in either parallel or sequential execution, but only on mac and windows. This small change can affect the second significant digit, which is significant. Curious to hear your thoughts about that.

Will implement TextTester soon.